File: public/docs/sdks/java/src/main/java/com/lydian/sdk/LydianIQClient.java
package com.lydian.sdk;

import com.lydian.sdk.models.Signal;
import com.lydian.sdk.models.KnowledgeEntity;
import com.lydian.sdk.models.Insight;

import java.io.IOException;

/**
 * Client for LyDian IQ API
 */
public class LydianIQClient {
    private final LydianClient client;

    public LydianIQClient(LydianClient client) {
        this.client = client;
    }

    public Signal ingestSignal(Signal signal) throws IOException {
        return client.request("POST", "/lydian-iq/signals", signal, Signal.class);
    }

    public KnowledgeEntity createEntity(KnowledgeEntity entity) throws IOException {
        return client.request("POST", "/lydian-iq/knowledge/entities", entity, KnowledgeEntity.class);
    }
}

File: Desktop/ailydian-ultra-pro/docs/sdks/csharp/LydianIQClient.cs
using System.Threading;
using System.Threading.Tasks;
using Lydian.SDK.Models;

namespace Lydian.SDK
{
    public class LydianIQClient
    {
        private readonly LydianClient _client;

        public LydianIQClient(LydianClient client)
        {
            _client = client;
        }

        public async Task<Signal> IngestSignalAsync(Signal signal, CancellationToken cancellationToken = default)
        {
            return await _client.RequestAsync<Signal>("POST", "/lydian-iq/signals", signal, cancellationToken)
                ?? throw new LydianException("Failed to ingest signal", 500);
        }

        public async Task<KnowledgeEntity> CreateEntityAsync(KnowledgeEntity entity, CancellationToken cancellationToken = default)
        {
            return await _client.RequestAsync<KnowledgeEntity>("POST", "/lydian-iq/knowledge/entities", entity, cancellationToken)
                ?? throw new LydianException("Failed to create entity", 500);
        }
    }
}
